src/goto-symex/memory_model.cpp

```diff
@@ -120,3 +94,38 @@ void memory_model_baset::read_from(symex_target_equationt &equation)
     }
   }
 }
+
+symbol_exprt memory_model_baset::register_read_from_choice_symbol(
+  const event_it &r,
+  const event_it &w,
+  symex_target_equationt &equation)
+{
+  bool is_rfi=
+    w->source.thread_nr==r->source.thread_nr;
```

Comment:
While you are at it: I'd move this further down, just right before the first use of `is_rfi`.

Reply:
Done